src/Mod/MeshPart/App/AppMeshPartPy.cpp

```diff
@@ -509,7 +512,14 @@ class Module : public Py::ExtensionModule<Module>
                 }
                 mesher.setColors(colors);
             }
-            return Py::asObject(new Mesh::MeshPy(mesher.createMesh()));
+            Mesh::MeshObject* mesh;
+            if (Base::asBoolean(releaseGIL)) {
+                Base::PyGILStateRelease();
+                mesh = mesher.createMesh();
+            } else {

```

Comment:
You must to give the temporary variable a name. With `Base::PyGILStateRelease();` the object is created and destroyed so the GIL is released and restored immediately.
Change the line to `Base::PyGILStateRelease releaser;` to release the GIL throughout the conditional block.